### PR TITLE
feat(distribution): add Nix Flake based duckdb shared library + jar output

### PR DESCRIPTION
**Pull request by thenonameguy**
Hey, thanks for the great library!

For ease of use, I added a Nix-based slim jar distribution method for my own purposes.
Would be happy to upstream the change, if you don't think it pollutes the repo too much. ✌️

### CODE REVIEW

File: src/tmducken/duckdb.clj [hunk 1]
```diff
@@ -105,14 +105,14 @@ _unnamed [5 3]:
             (when-not is-init?
               (let [duckdb-home (or duckdb-home
                                     (System/getenv "DUCKDB_HOME")
-                                    "./binaries")
+                                    (str "/" com.sun.jna.Platform/RESOURCE_PREFIX))
```
Comment (thenonameguy) [inline src/tmducken/duckdb.clj:108]: This is a breaking change, but probably only for local environments.
With the other changes in this commit, if you use `nix shell` or `direnv`, the duckdb library gets put in the correct place automatically, so it doesn't break any existing development workflows.
The binaries folder could be removed then.

File: flake.nix [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+                projectSrc = ./.;
+                name = "com.techascent/tmducken";
+                buildCommand = "clj -T:build jar --report stderr";
+                preBuild = "export DUCKDB_VERSION=${version}";
+              };
+        };
+        devShells.default = pkgs.mkShell {
+          buildInputs = [
+            # provides deps-lock library for updating deps-lock.json from deps.edn
+            clj-nix.packages.${system}.deps-lock
```
Comment (thenonameguy) [inline flake.nix:47]: `deps-lock` needs to be called to produce a lock file so `nix build` is truly repeatable.

File: README.md [hunk 3]
```diff
@@ -97,6 +97,26 @@ user in data form there is no advantage to using apache arrow as a query result.
 
 ## Usage
 
+### Preferred method (Nix)
+
+You can get a jar for your own OS (Linux/MacOS) with duckdb included by [installing Nix](https://github.com/DeterminateSystems/nix-installer),
+then running:
+
+``` console
+$ nix build github:schemamap/tmducken
```
Comment (thenonameguy) [inline README.md]: If you approve the merging of this PR, will update this to reflect the correct org, post-merge.

File: src/tmducken/duckdb.clj (hunk 1 above)
Comment (thenonameguy) [reply src/tmducken/duckdb.clj:108]: It loads like this now:
```
INFO: Attempting to load duckdb from "/darwin-aarch64/libduckdb.dylib"
```

Comment (cnuernber) [reply src/tmducken/duckdb.clj:108]: I love it - we will want to work this in for sure.  I think it is fine actually as local devs can be sure to set DUCKDB_HOME - we always are.

File: README.md (hunk 3 above)
Comment (cnuernber) [reply README.md]: Approved!!

File: flake.nix (hunk 2 above)
Comment (thenonameguy) [reply flake.nix:47]: Added a README.md note for this.

**Comment by cnuernber** [comment]
This is great - people like to have the binaries in the jar as it simplifies other things.  I'm going to merge this assuming it doesn't break other things such as the existing scripts/deploy pathway.  My guess is you will often have to rebuild the lockfile as we aren't currently using nix here.

**Comment by thenonameguy** [comment]
Thanks for merging! 🙏 

There actually exists an automated solution via Github Actions that opens a PR to keep the lockfile in sync:
https://jlesquembre.github.io/clj-nix/github-action/

I'll use the this repo now, instead of my fork. 
I'll open a PR next time I upgrade it so the above is set up, without much manual intervention.

**Pull request merged**